src/index.tsx

```diff
@@ -669,17 +696,26 @@ class DraggableFlatList<T> extends React.Component<Props<T>, State> {
             and(
               this.isAutoscrolling.native,
               or(
+                // We've scrolled to where we want to be
                 lessOrEq(
                   abs(sub(this.targetScrollOffset, this.scrollOffset)),
                   scrollPositionTolerance
                 ),
-                this.isScrolledUp,
-                this.isScrolledDown
+                // We're at the start, but still want to scroll farther up
+                and(
+                  this.isScrolledUp,
+                  lessOrEq(this.targetScrollOffset, this.scrollOffset)
+                ),
+                // We're at the end, but still want to scroll further down
+                and(
+                  this.isScrolledDown,
+                  greaterOrEq(this.targetScrollOffset, this.scrollOffset)
+                )
               )
             ),
             [
+              // Finish scrolling
               set(this.isAutoscrolling.native, 0),
-              this.checkAutoscroll,
```

Comment:
why was this removed? was it causing a bug or simply not necessary?

Reply:
From what I understand, this was used before when `autoscroll` was a recursive function and resulted in duplicate calls to `autoscroll`. It can be kept in too to minimize changes!

The past code:

- `onScroll` would be called, and both call `this.checkAutoscroll` and resolve `this.onAutoscrollComplete`
  - `this.checkAutoscroll` would call `this.autoscroll` with `this.autoscrollParams`
  - `onAutoscrollComplete` would be called with `this.autoscrollParams`, which would resolve the promise that line 657 is awaiting `const nextScrollParams = await this.scrollToAsync(targetOffset);`
  - The end result is that on line 658, `this.autoscroll` would be called again with `this.autoscrollParams`

The end result is we just remove this duplicate call to `this.autoscroll`.